bootstrap/src/main/java/com/splunk/opentelemetry/javaagent/bootstrap/metrics/ThreadPoolSemanticConventions.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.javaagent.bootstrap.metrics;
+
+import static com.splunk.opentelemetry.javaagent.bootstrap.metrics.GaugeSemanticConvention.gauge;
+import static io.micrometer.core.instrument.binder.BaseUnits.TASKS;
+import static io.micrometer.core.instrument.binder.BaseUnits.THREADS;
+
+public final class ThreadPoolSemanticConventions {
+
+  /** The current number of threads in the pool. */
+  public static final GaugeSemanticConvention THREADS_CURRENT = gauge("executor.threads", THREADS);
+  /** The number of threads that are currently busy. */
+  public static final GaugeSemanticConvention THREADS_ACTIVE =
+      gauge("executor.threads.active", THREADS);
+  /** The number of threads that are currently idle. */
+  public static final GaugeSemanticConvention THREADS_IDLE =
+      gauge("executor.threads.idle", THREADS);
+  /** Core thread pool size - the number of threads that are always kept in the pool. */
+  public static final GaugeSemanticConvention THREADS_CORE =
+      gauge("executor.threads.core", THREADS);
+  /** The maximum number of threads in the pool. */
+  public static final GaugeSemanticConvention THREADS_MAX = gauge("executor.threads.max", THREADS);
+  /** The total number of tasks that were submitted to this executor. */
+  public static final GaugeSemanticConvention TASKS_SUBMITTED =
+      gauge("executor.tasks.submitted", TASKS);
```

Comment:
Ah ok, Watson and I chatted about this and it looks like the pool just tracks a singular number that happens to be monotonically increasing but does not report deltas...so we'd have to keep state and do that math ourselves each time, which isn't great. I'm sure this is fine as a gauge.  Very minor (microscopic) bummer.

Reply:
I thought that too, but I checked the micrometer API once again and it turns out there's a `FunctionCounter` instrument there that can keep track of a monotonically increasing function - which fits the purpose of those `TASKS_*` meters. So I ended up using a `FunctionCounter` instead of a gauge after all.
(thanks for the comment by the way - I probably wouldn't have ended up scanning through the micrometer API if it weren't for that)